splice_machine/src/main/java/com/splicemachine/derby/impl/storage/SpliceRegionAdmin.java

```diff
@@ -89,6 +92,290 @@
+        columnInfo[3]=new GenericColumnDescriptor("OWNING_SERVER",
+                DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR,32672));
+
+        ArrayList<ExecRow> rows=new ArrayList<>();
+        DataValueDescriptor[] dvds=new DataValueDescriptor[]{
+                new SQLVarchar(),
+                new SQLVarchar(),
+                new SQLVarchar(),
+                new SQLVarchar()
+        };
+
+        List<Partition> partitions =  table.subPartitions(new byte[0], new byte[0], true);
+        for (Partition p : partitions) {
+            ExecRow row=new ValueRow(dvds.length);
+            row.setRowArray(dvds);
+            String serverName = getServerName(p.owningServer().getHostname(),
+                    p.owningServer().getPort(),
+                    p.owningServer().getStartupTimestamp());
+            dvds[0].setValue(p.getEncodedName());
+            dvds[1].setValue(Bytes.toStringBinary(p.getStartKey()));
+            dvds[2].setValue(Bytes.toStringBinary(p.getEndKey()));
+            dvds[3].setValue(serverName);
+            rows.add(row.getClone());
+        }
+
+        EmbedConnection defaultConn=(EmbedConnection) SpliceAdmin.getDefaultConn();
+        Activation lastActivation=defaultConn.getLanguageConnection().getLastActivation();
+        IteratorNoPutResultSet resultsToWrap=new IteratorNoPutResultSet(rows,columnInfo,lastActivation);
+        resultsToWrap.openCore();
+        results[0] = new EmbedResultSet40(defaultConn,resultsToWrap,false,null,true);
+    }
+
+    public static final String SERVERNAME_SEPARATOR = ",";
+    static String getServerName(String hostName, int port, long startcode) {
+        final StringBuilder name = new StringBuilder();
+        name.append(hostName);
+        name.append(SERVERNAME_SEPARATOR);
+        name.append(port);
+        name.append(SERVERNAME_SEPARATOR);
+        name.append(startcode);
+        return name.toString();
+    }
+
+    private static long getConglomerateNumber(LanguageConnectionContext lcc, String schemaName, String objectName) throws SQLException, StandardException {
+        TransactionController tc = lcc.getTransactionExecute();
+        DataDictionary dd = lcc.getDataDictionary();
+        SchemaDescriptor sd = dd.getSchemaDescriptor(schemaName, tc, true);
+        TableDescriptor td = dd.getTableDescriptor(objectName, sd, tc);
+        long conglomerateNumber = 0;
+        if (td != null) {
+            conglomerateNumber = td.getHeapConglomerateId();
+        }
+        else {
+            ConglomerateDescriptor cd = dd.getConglomerateDescriptor(objectName, sd, false);
+            if (cd == null) {
+                throw StandardException.newException(SQLState.LANG_TABLE_NOT_FOUND, schemaName);
+            }
+            conglomerateNumber = cd.getConglomerateNumber();
+        }
+
+        return conglomerateNumber;
+    }
+    /**
+     *
+     * @param schemaName
+     * @param objectName
```

Comment:
this objectName seems to be rather tableName, no? please rename or provide descriptive comment.

Reply:
so ASSIGN_TO_SERVER is only moving the table itself, but not its indices? please explain in comment.
i guess you would do a ASSIGN_TO_SERVER, then LOCALIZE_INDEXES_FOR_TABLE, right?

Reply:
ASSIGN_TO_SERVER can move an index to another region server